docs/index.rst

```diff
@@ -1,14 +1,207 @@
+
+.. warning::
+
+   Are you not sure if you are using this command in a right way?  No
+   problem! The test-case command can be used with ``--collect-only`` option
+   which runs in a dry run mode and shows the changes it would have made
+   without actually making them:
+
+     .. code-block:: bash
+
+         $ betelgeuse test-case --path sample_project/tests/ PROJECT_CLOUD \
+         --collect-only
+
+         Creating test case test_login_1 for requirement: Login Example.
+         Linking test case test_login_1 to requirement: Login Example.
+
+.. _test_results_command:
+
+test-results command
+++++++++++++++++++++
+Gives a nice summary of test cases/results in the given jUnit XML file.
+
+.. code-block:: bash
+
+    $ betelgeuse test-results --path \
+    sample_project/results/sample-junit-result.xml
+
+    Passed: 1
+
+.. _test_run_command:
+
+test-run command
+++++++++++++++++
+Creates a test run in polarion using the information in the given jUnit XML
+file. This command performs the following steps:
+
+- Parses the jUnit XML file to read all the test cases and their run statuses.
+- Creates a new test run and updates it with all the parsed test case items and
+  their run statuses.
+
+.. code-block:: bash
+
+    $ betelgeuse test-run --path sample_project/results/sample-junit-result.xml \
+    --test-run-id regression_test_run_1 --test-template-id Empty --user \
+    testuser1 --source-code-path sample_project/tests/ PROJECT_CLOUD
+
+    Test run regression_test_run_1 found.
+    Adding test record for test case PROJECT_CLOUD-12655 with status passed.
+
+At this time, it is very important to understand how Betelgeuse links the items
+in the jUnit XML report to the actual source code.  To help in this process,
+it is a must that both the test runner and Betelgeuse get called in the same
+directory.  Consider the following jUnit XML report which just has one test case
+for easy understanding:
+
+.. code-block:: xml
+
+    <testcase classname="sample_project.tests.test_login_example.LoginTestCase"
+    file="sample_project/tests/test_login_example.py" line="421" name="test_login_1"
+    time="694.768339396">...</testcase>
+
+With the above report, Betelgeuse performs the following:
+
+- Derives the test method by joining its ``classname`` and ``name`` attributes
+  with a dot.  In this case, it becomes 
+  ``sample_project.tests.test_login_example.LoginTestCase.test_login_1``.
+- Looks at the ``--source-code-path`` option value and converts every test
+  module path into a python import path.  For example,
+  ``sample_project/tests/test_login_example.py`` will become
+  ``sample_project.tests.test_login_example``.  All test methods or functions are
+  then appended.  For example, the method ``test_login_1`` inside the class
+  ``LoginTestCase`` will be generated as
+  ``sample_project.tests.test_login_example.LoginTestCase.test_login_1``.
+- The information obtained from both the steps above are compared and ``@ID``
```

Comment:
This is explaining the code logic in the same way as it is written.  Assumption is that this needs an in-depth understanding of how betelgeuse works.  Can you type some examples of how you think this can be improved?